azurerm/internal/services/keyvault/resource_arm_key_vault.go

```diff
@@ -408,6 +457,33 @@ func resourceArmKeyVaultDelete(d *schema.ResourceData, meta interface{}) error {
 		}
 	}
 
+	if meta.(*clients.Client).Features.KeyVault.PurgeSoftDeleteOnDestroy && softDeleteEnabled {
+		location := d.Get("location").(string)
+
+		// raise an error if the key vault is in the soft delete state and purge protection is enabled
+		if purgeProtectionEnabled {
+			delDate, purgeDate, err := azure.KeyVaultGetSoftDeletedState(ctx, client, name, location)
+			if err == nil {
+				if delDate != nil && purgeDate != nil {
+					return fmt.Errorf("unable to purge Key Vault %q (Resource Group %q) because it has purge protection enabled. The key vault was soft deleted on %s and is scheduled to be purged on %s", name, resourceGroup, delDate.(string), purgeDate.(string))
+				}
+				return fmt.Errorf("unable to purge Key Vault %q (Resource Group %q) because it has purge protection enabled", name, resourceGroup)
+			}
+		}
+
+		log.Printf("[DEBUG] KeyVault %s marked for purge, executing purge", name)
+		future, err := client.PurgeDeleted(ctx, name, location)
+		if err != nil {
+			return err
+		}
+
+		log.Printf("[DEBUG] Waiting for purge of KeyVault %s", name)
+		err = future.WaitForCompletionRef(ctx, client.Client)
+		if err != nil {
+			return fmt.Errorf("Error purging Key Vault %q (Resource Group %q): %+v", name, resourceGroup, err)
+		}
+	}
```

Comment:
since we can't delete this for ~90 days we should probably remove this behaviour, unfortunately

if/when the API supports it however we should look to add this back in

Reply:
There is an edge case where we can delete the key vault, if the key vault has soft delete enabled, but has never enabled purge protection we can totally delete the key vault, however once purge protection is enabled, you are correct we will have to wait the 90 days before the key vault gets automatically permanently deleted by Azure.